File: java/lib/src/main/java/com/svix/models/MessageStatusText.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.Utils.ToQueryParam;

public enum MessageStatusText implements ToQueryParam {
    SUCCESS("success"),
    PENDING("pending"),
    FAIL("fail"),
    SENDING("sending");
    private final String value;

    MessageStatusText(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: csharp/Svix/Models/MessageStatusText.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Svix.Models
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum MessageStatusText
    {
        [EnumMember(Value = "success")]
        Success = 1,

        [EnumMember(Value = "pending")]
        Pending = 2,

        [EnumMember(Value = "fail")]
        Fail = 3,

        [EnumMember(Value = "sending")]
        Sending = 4,
    }
}
